### ISSUE TITLE
[Feature Request] Ability to use RETURNING in SQLite - Bump SQLite Version up to 3.35.0

### ISSUE THREAD

**Issue by mcunningham** [prisma/prisma#6876]
## Problem

Opening up an Feature request after a quick thread I had with @janpio in the prisma slack channel.

I have a use case where I need to perform many upserts (1k - 20k) into a SQLite table and return the rows that were updated/inserted, however as opposed to a normal prisma [upsert](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update-or-create-records), I need to **only** process the update and return the value if the a specific value in the row for a given upsert is **not** the same as the previous.  Since this requires comparing current database values with the incoming request, which I don't believe is supported in Prisma ([see related here](https://github.com/prisma/prisma/discussions/2531)), I begun to reach for [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) which can accomplish this with something similar to the below.

```
INSERT INTO phonebook(name,phonenumber,validDate)
  VALUES('Alice','704-555-1212','2018-05-08')
  ON CONFLICT(name) DO UPDATE SET
    phonenumber=excluded.phonenumber,
    validDate=excluded.validDate
  WHERE excluded.phonenumber!=phonebook.phonenumber RETURNING *;
```
_Note: This is a simplified example edited directly from [SQLite docs](https://www.sqlite.org/lang_UPSERT.html) with an added RETURNING * statement at the end. (Not meant to be run but shows the use case)_

In this example, a failing WHERE clause would result the DO UPDATE to becoming a no-op. (ie. If the old phone number in the database was the same as the one being inserted, then no upsert would occur)

My plan was to try and run all of these custom $queryRaw upserts in a prisma [$transaction ](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#the-transaction-api) and then receive the ones that successfully were upserted (passed the WHERE clause). Which I think will work. However the problem now is that [RETURNING](https://sqlite.org/lang_returning.html) was introduced to SQLite in version 3.35.0, and the SQLite version in my prisma client is 3.31.1, so I cannot use this as a solution until the SQLite version in the prisma client is at least 3.35.0.

## Suggested solution

I am requesting either giving the user the ability to upgrade SQLite version, or bumping up the fixed version of SQLite that is shipped in the prisma client to 3.35.0 or higher so that RETURNING can be used in raw queries. 

## Alternatives

I have looked into potential workarounds using [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware). However, I do not believe there is a good way to accomplish comparison of current database values to the incoming values using this method. Please correct me if I am wrong here. 

Thanks for all your hard work on such an awesome product!

**Comment by pimeys** [comment]
You can expect this in version 2.23.0 in a few weeks. The changes were pretty easy, but I noticed a few breaking changes in the CHANGELOG that might cause trouble, and I don't want to cause breakage so close to the release.

**Comment by mcunningham** [comment]
@pimeys No problem/rush on this if there are potential breaking changes and it needs more tests. For the meantime I found a temporary workaround using `$executeRaw` plus middleware where I am checking to see if the there were any affected rows and then returning from middleware the original params if so. Also, looks like SQLite  had some fixes to RETURNING in 3.35.4 according to their [Release History](https://sqlite.org/changes.html) so maybe something to keep in mind.

**Comment by pimeys** [comment]
Yeah what I have here in my PR is SQLite version `3.35.4`. It was a bit of a surprise the `RETURNING` doesn't work with temporary tables. I had a sample rust program with a properly working `RETURNING` statement, but our test was red due to not getting any data back from the database.

Found out that `RETURNING` returns nothing if the table is temporary...

What I'm now trying to see is how the rust sqlite driver changed some datetime string representations. I can still only generate integer date fields, but I'd like to find out a schema that actually stores dates as strings. All tests green in all our suites doesn't really help now... :P

Basically these changes worry me a bit:
- https://github.com/rusqlite/rusqlite/pull/886
- https://github.com/rusqlite/rusqlite/pull/928

**Comment by pimeys** [comment]
Now to wish that MySQL would implement `RETURNING` too, and we could use it in all SQL databases in the next set of performance iteration...

**Comment by mcunningham** [comment]
Got it, yeah looks like you are right about the the virtual tables in [3. Limitations And Caveats](https://sqlite.org/lang_returning.html) from SQLite docs states `RETURNING` clause is not available for `DELETE` and `UPDATE` against virtual tables. `RETURNING` is a relatively new feature in  SQLITE so I imagine some of the limitations will get worked out over time. I will leave this open to track further. I appreciate you looking into this so quickly!
